Summer school wrap-up

jngrad opened the pull request:
Description of changes:
- document broken CMake feature `make test` (closes  #4370)
- improve documentation of energy/pressure calculation (fixes #4369)
- document energy minimization with Steepest Descent (fixes #3898)
- modernize visualizer tutorial with Steepest Descent, tqdm and logging
- cleanup core and LB test cases to synchronize with the walberla branch
- fix broken tests on configurations not tested in CI (feedback from the summer school)

jngrad committed:
docs: Example for energy minimization

jngrad committed:
docs: Use steepest descent instead of force capping

jngrad committed:
docs: Modernize visualization tutorial

Simplify code, use logging and tqdm instead of print()
statements, use WCA potential, plot thermalization curve.

jngrad committed:
core: Cleanup comments, error messages, globals

jngrad committed:
tests: Cleanup LB test cases

jngrad committed:
docs: Update python package requirements

jngrad commented:
During today's online tech support session, participants found issues in the unit tests and python tests. I'll push the fixes here.

jngrad committed:
tests: Fix tests for CentOS and no-rotation builds

jngrad committed:
python: Fix observables documentation

biermanncarl commented:
I also found some minor things during my preparation for the tutorials. I considered making a separate pull-request, but could also make some changes on top of this branch.

jngrad commented:
@biermanncarl good idea, you can write your changes on top of my branch in your fork, and I'll forward them here. I have one more thing add to this PR (documenting something in the reaction methods), if you are faster than me I'll commit it after your changes.

biermanncarl committed:
Minor corrections

biermanncarl committed:
docs: Observables documentation maintenance and restructuring to make it less confusing

jngrad committed:
docs: Cleanup Sphinx code examples

Document reaction method slab geometry with an example and remove
particle ids from code samples. Document broken `make test` feature.

christophlohrmann commented on doc/sphinx/analysis.rst:
```diff
@@ -578,17 +636,22 @@ Example: Calculating a particle's diffusion coefficient
 
 The full example can be found in :file:`samples/diffusion_coefficient.py`.
+Note that in this example, the operation :literal:`square_distance_componentwise`
+is used, which is not a true correlation function but rather the squared
```
Why are we so critical of the poor MSD correlator?

jngrad replied:
Should be fixed by 24801f50825.

christophlohrmann commented on doc/sphinx/installation.rst:
```diff
@@ -87,9 +87,9 @@ are required:
 .. code-block:: bash
 
-    sudo apt install python3-matplotlib python3-pint ipython3 jupyter-notebook
+    sudo apt install python3-matplotlib python3-pint python3-tqdm ipython3 jupyter-notebook
```
why do we not install pint and tqdm via pip?

jngrad replied:
We do not want to use pip in general, because users end up with a complex mix of package versions. This can be a source of frustration when running samples and tutorials that require a different version of the packages. Installing packages from Ubuntu repositories ensures that everyone has the same versions, that packages are compatible with each other and that they remain in sync with the NEP 29 schedule.

There are a few packages that need to be installed via pip. For example `scipy` is too old in the Ubuntu repository and is missing the rotation matrix features that we use in python tests. Same issue with Sphinx, which doesn't contain features that are crucial to the ESPResSo user guide. There used to be a time where `pint` also had to be installed from pip because of a critical bug when used with `numpy.cbrt()`.

jngrad commented on doc/sphinx/analysis.rst:
```diff
@@ -578,17 +636,22 @@ Example: Calculating a particle's diffusion coefficient
+is used, which is not a true correlation function but rather the squared
+distance of each component to the first sample is being computed. Other
+available operations include actual correlation functions, as described
+in the source documentation of :class:`espressomd.accumulators.Correlator`.
```
```suggestion
Note that in this example, the operation :literal:`square_distance_componentwise`
is used, which is not a correlation function in the mathematical sense. Other
available operations include actual correlation functions, as described
in the source documentation of :class:`espressomd.accumulators.Correlator`.
```

jngrad committed:
docs: Rephrase MSD description

Co-authored-by: Carl Georg Biermann <carlgeorg@gmx.net>
